### PR TITLE
SR-10619: Order in which DateFormatter settings are set up matters on Linux

### PR DESCRIPTION
**Pull request by spevans**
- Ensure the setting of .dateStyle and .timeStyle isnt dependant on the
  current setting of .locale.

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci test

**Comment by spevans** [comment]
cc @weissi

File: TestFoundation/TestDateFormatter.swift [hunk 1]
```diff
@@ -443,4 +444,64 @@ class TestDateFormatter: XCTestCase {
...
+             (".dateStyle", {
+                $0.dateStyle = .medium
+             }),
+             (".timeStyle", {
+                $0.timeStyle = .medium
+             })
+        ]
+
+        // Test all of the combinations of setting the properties produces the same output
+        let expected = "5. mai 2019, 12:52:10"
```
Comment (millenomi) [inline TestFoundation/TestDateFormatter.swift]: Can this be like other tests in that we do not hardcode an answer, so we avoid ICU changes from causing breakage?

Comment (spevans) [reply TestFoundation/TestDateFormatter.swift]: I tested this against Darwin Foundation and it worked find so I don't think this string is ICU version dependant. The PR actually fixes all of the current tests that are disabled due to not working correctly versus Darwin and will be enabled in a followup PR.

**Pull request merged**